code/FileAttachmentFieldCleanTask.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * Delete all files being tracked that weren't saved against anything.
+ *
+ * WARNING: You must call 'FileAttachmentFieldTrack::untrack' against IDs on custom-built forms or you
+ *          -will- remove files accidentally.
+ *
+ * @package  unclecheese/silverstripe-dropzone
+ */
+class FileAttachmentFieldCleanTask extends BuildTask {
+    protected $title = "File Attachment Field - Clear all tracked files that are older than 1 hour";
+    
+    protected $description = 'Delete files uploaded via FileAttachmentField that aren\'t attached to anything.';
+    
+    public function run($request) {
+        $files = FileAttachmentFieldTrack::get()->filter(array('Created:LessThanOrEqual' => date('Y-m-d H:i:s', time()-3600)));
+        $files = $files->toArray();
+        if ($files) {
+            foreach ($files as $trackRecord) {
+                $file = $trackRecord->File();
+                if ($file && $file->exists()) {
```

Comment:
I was just speculating about a breaking change in 4.0. 
I'll make this change.

My rationale was more about factoring/moving slices of code between different contexts (ie. using a has_one or retrieving a record via DataList->first())